ts/visitor.ts

```diff
@@ -357,14 +357,10 @@ export const visitor = {
   },
   LogicalExpression: { // logical expressions only has && and || as operators
     exit(path: NodePath<t.LogicalExpression>, st: S) {
-      let op = path.node.operator;
-      let opName = t.stringLiteral(op);
-      path.replaceWith(dynCheck(
-        'applyBinaryBooleanOp',
-        path.node.loc,
-        opName,
-        path.node.left,
-        path.node.right, 
+      path.replaceWith(t.logicalExpression(
+        path.node.operator,
+        dynCheck('checkIfBoolean', path.node.left.loc, path.node.left),
```

Comment:
If you pass in `path.node.operator` as an argument to the dynCheck, you can print the same format error message as before, including which operator this check resulted from.